lightwood/encoders/encoder_base.py

```diff
@@ -0,0 +1,19 @@
+
+class EncoderBase:
+    """Base class for all encoders"""
+    def __init__(self, is_target=False):
+        self.is_target = is_target
+
+    def prepare_encoder(self, priming_data):
+        """Can be overridden"""
```

Comment:
I think the comments for "Can be overdeen" aren't necessary, we can just leave the `pass` there and that's indication enough.

Reply:
Also I think `prepare_encoder` should probably raise `NotImplementedError` , since all encoders should be prepared.

Reply:
I think any encoder defines for itself whether it has to be prepared or not, so it's a virtual method.
.encode() and .decode() **must** be implemented because they're expected to return specific type of data (e.g. list of vectors in case of encode).

Reply:
Ah, yes, you're correct there.